Library/Homebrew/formula_auditor.rb

```diff
@@ -865,14 +872,19 @@ def head_only?(formula)
     end
 
     def linux_only_gcc_dep?(formula)
-      # TODO: Make this check work when running on Linux and not simulating macOS too.
-      return false unless Homebrew::SimulateSystem.simulating_or_running_on_macos?
+      odie "`#linux_only_gcc_dep?` works only on Linux!" if Homebrew::SimulateSystem.simulating_or_running_on_macos?
 
       formula_hash = formula.to_hash_with_variations
-      deps = formula_hash["dependencies"]
-      linux_deps = formula_hash.dig("variations", :x86_64_linux, "dependencies")
+      linux_deps = formula_hash["dependencies"]
+      variations_deps = []
+      formula_hash["variations"].map do |variation, data|
```

Comment:
Maybe I need to be checking this `variations` hash for each of `MacOSVersions::SYMBOLS.keys`.

Reply:
Oh, but doing this ignores the `arm64_*` keys.